pyanaconda/core/glib.py

```diff
@@ -46,16 +46,43 @@
            "MAXUINT"]
 
 
-def create_main_loop():
+def create_main_loop(mainctx=None):
     """Create GLib main loop.
 
+    :param mainctx: create a mainloop with mainctx
     :returns: GLib.MainLoop instance.
     """
-    return MainLoop()
+    return MainLoop(mainctx)
 
 
 def create_new_context():
     """Create GLib context.
 
     :returns: GLib.MainContext."""
     return MainContext.new()
+
+
+def mainloop_run(timeout=0, mainloop=None):
```

Comment:
Please add doc text with specification of the timeout units.

Reply:
ok，I will fix this

Reply:
Addressed in https://github.com/rhinstaller/anaconda/pull/4322